virtual_ecosystem/models/animal/animal_cohorts.py

```diff
@@ -299,8 +381,17 @@ def get_eaten(
         # Update the number of individuals in the prey cohort
         self.individuals -= actual_individuals_killed
 
+        # set cohort to not alive if all the individuals are dead
+        if self.individuals <= 0:
+            self.is_alive = False
+
+        # Find the intersection of prey and predator territories
+        intersection_carcass_pools = self.territory.find_intersecting_carcass_pools(
+            predator.territory, carcass_pools
+        )
+
         # Update the carcass pool with carcass mass
-        self.update_carcass_pool(carcass_mass, carcass_pool)
+        self.update_carcass_pool(carcass_mass, intersection_carcass_pools)
```

Comment:
Why do you need to update the carcass pool with the intersection of carcass pools instead of the normal carcass pool, as before? Indeed, why is the intersection of territories needed and why is it calculated based on the intersection of the carcass pools?